base_classes/NXcalibration.nxdl.xml

```diff
@@ -0,0 +1,220 @@
+                   * A variable name cannot start with a number
+                   * A variable name can only contain alpha-numeric characters and underscores (A-z, 0-9, and _ )
+                   * Variable names are case-sensitive (age, Age and AGE are three different variables)
+            </doc>
+        </attribute>
+        <attribute name="input_path">
+            <doc>
+                 The path from which this data is derived, e.g., raw detector axis.
+                 Should be a valid NeXus path name, e.g., /entry/instrument/detector/raw.
+            </doc>
+        </attribute>
+    </field>
+    <field name="input_SYMBOL" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             Additional input axis to be used in the formula.
+             The part after `input_` is used as the symbol to be used in the `fit_function`, i.e.,
+             if the field name is `input_my_field` you should refer to this axis by `my_field` in the `fit_function`.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ncal"/>
+        </dimensions>
+        <attribute name="input_path">
+            <doc>
+                 The path from which this data is derived, e.g., raw detector axis.
+                 Should be a valid NeXus path name, e.g., /entry/instrument/detector/raw.
+            </doc>
+        </attribute>
+    </field>
+    <field name="coefficients" type="NX_FLOAT" units="NX_ANY">
+        <doc>
+             For non-linear energy calibrations, e.g. in a TOF, a polynomial function is fit
+             to a set of features (peaks) at well defined energy positions to determine
+             E(TOF). Here we can store the array of fit coefficients.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="ncoeff"/>
+        </dimensions>
+    </field>
+    <field name="fit_function" type="NX_CHAR">
+        <doc>
+             For non-linear energy calibrations. Here we can store the formula of the
+             fit function.
+             
+             Use a0, a1, ..., an for the coefficients, corresponding to the values in the coefficients field.
+             
+             Use x0, x1, ..., xn for the nth position in the `original_axis` field.
+             If there is the symbol attribute specified for the `original_axis` this may be used instead of x.
+             If you want to use the whole axis use `x`.
+             Alternate axis can also be available as specified by the `input_SYMBOL` field.
+             The data should then be referred here by the `SYMBOL` name, e.g., for a field
+             name `input_my_field` it should be referred here by `my_field` or `my_field0` if
+             you want to read the zeroth element of the array.
+             
+             The formula should be numpy compliant.
+        </doc>
+    </field>
+    <field name="scaling" type="NX_FLOAT" units="NX_ANY">
```

Comment:
Use scaling_factor and grab the docstrings for scaling_factor and offset from NXdata

Reply:
Would mean correcting the formula to the one used in NXdata